src/js/structs/List.js

```diff
@@ -66,6 +66,26 @@ module.exports = class List {
   }
 
   /**
+   * Return a new list with the items currently present in this list and
+   * the list on the first argument, without duplicates.
+   *
+   * @param {List} list - The list to combine with this list
+   * @returns {List} Returns a new list with the combined items
+   */
+  combine(list) {
+    let myItems = this.getItems();
+    let newItems = list.reduceItems(function (items, item) {
+      if (myItems.indexOf(item) === -1) {
```

Comment:
Yeah, I thought about ways to optimise this, but Object won't work, since `item`s are objects and object keys can only be strings. A `Map` could be used for what you suggest, however I am not sure it's widely supported yet.